### ISSUE TITLE
 Failed to download resource "rsync--patch"

### ISSUE THREAD

**Issue by fzalkow** [jedda/homebrew-counterpart#1]
Hi,

thanks for counterpart, seems like a very interesting project that I would like to use.

But I cannot install its dependency:

    ==> Installing counterpart from jedda/homebrew-counterpart
    /usr/bin/curl -fLA Homebrew 0.9.5 (Ruby 2.0.0-481; Mac OS X 10.9.5) https://raw.github.com/Homebrew/homebrew-dupes/109dca908c6499116e07483d7afe8a1c3ef63ad6/rsync.rb -o /Library/Caches/Homebrew/Formula/rsync.rb
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
      0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    100   829  100   829    0     0    913      0 --:--:-- --:--:-- --:--:--  809k
    ==> Installing counterpart dependency: https://raw.github.com/Homebrew/homebrew-dupes/109dca908c6499116e07483d7afe8a1c3ef63ad6/rsync.rb
    ==> Downloading http://rsync.samba.org/ftp/rsync/src/rsync-3.0.9.tar.gz
    Already downloaded: /Library/Caches/Homebrew/rsync-3.0.9.tar.gz
    ==> Verifying rsync-3.0.9.tar.gz checksum
    tar xf /Library/Caches/Homebrew/rsync-3.0.9.tar.gz
    ==> Downloading http://gitweb.samba.org/?p=rsync-patches.git;a=blob_plain;hb=v3.0.9;f=fileflags.diff
    /usr/bin/curl -fLA Homebrew 0.9.5 (Ruby 2.0.0-481; Mac OS X 10.9.5) http://gitweb.samba.org/?p=rsync-patches.git;a=blob_plain;hb=v3.0.9;f=fileflags.diff -C 0 -o /Library/Caches/Homebrew/rsync--patch-http%3A%2F%2Fgitweb.samba.org%2F%3Fp%3Drsync-patches.git%3Ba%3Dblob_plain%3Bhb%3Dv3.0.9%3Bf%3Dfileflags.diff.diff.incomplete
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
    100   357  100   357    0     0    817      0 --:--:-- --:--:-- --:--:--   816
      0     0    0 49261    0     0   8505      0 --:--:--  0:00:05 --:--:--     0
    curl: (18) transfer closed with outstanding read data remaining
    Error: Failed to download resource "rsync--patch"
    Download failed: http://gitweb.samba.org/?p=rsync-patches.git;a=blob_plain;hb=v3.0.9;f=fileflags.diff

If I call just `brew install rsync` it works but that doesn't satisfy the counterpart-dependency...

Any help is appreciated.

**Comment by timfenney** [comment]
AFAICT, this is now failing due to curl not being able to handle redirects to https.

Homebrew will warn you about the dangers of replacing it, but you can get curl from the `dupes` tap. I did:
```
brew tap homebrew/dupes
brew install curl --with-rtmpdump \
    --with-openssl \
    --with-libssh2 \
    --with-nghttp2 \
    --with-libmetalink \
    --with-libidn \
    --with-c-ares
```

But I believe the key is the `--with-openssl` option. This made the error go away for me.

**Comment by timfenney** [comment]
Of course, someone will have to figure out what kind of urls are acceptable for Homebrew recipes, and what kind of curl Homebrew should require. If the urls are ok as is, then Homebrew should come with a custom-built curl...
...so I would tend to see this as a workaround until a more suitable upstream url is found.

**Comment by fzalkow** [comment]
Didn't work for me... To be sure I even replaced my standard curl:

	$ sudo mv /usr/bin/curl /usr/bin/curl.bak
	$ sudo ln -s /usr/local/Cellar/curl/7.44.0/bin/curl /usr/bin/curl
	$ brew install counterpart -verbose

But I get the same error as above...

**Comment by timfenney** [comment]
Hmm, maybe we have a different recipe, I may have edited mine with `brew edit counterpart`. I have:

```
require 'formula'

class Counterpart < Formula
  homepage 'http://jedda.me/counterpart'
  url 'https://github.com/jedda/Counterpart/tarball/3bfe006078874d001681d2354892d93cca1d1705/counterpart-1.2.1.tar.gz'
  sha1 '98dbdf7e21f491d8bf7012af0ed05630c57c922b'

  depends_on 'https://raw.githubusercontent.com/Homebrew/homebrew-dupes/1663b8f234a0b8f6d376572dfe25e6c1f18066d9/rsync.rb'

  def install
    (bin).install "counterpart"
  end
end
```

**Comment by fzalkow** [comment]
Still an error with `rsync--patch`...

	$ brew install counterpart -verbose
	==> Installing counterpart from jedda/homebrew-counterpart
	/usr/bin/curl -fLA Homebrew 0.9.5 (Ruby 2.0.0-481; OS X 10.9.5) https://raw.githubusercontent.com/Homebrew/homebrew-dupes/1663b8f234a0b8f6d376572dfe25e6c1f18066d9/rsync.rb -o /Library/Caches/Homebrew/Formula/rsync.rb
	  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
	                                 Dload  Upload   Total   Spent    Left  Speed
	100  1702  100  1702    0     0   4026      0 --:--:-- --:--:-- --:--:--  4233
	==> Installing dependencies for jedda/counterpart/counterpart: https://raw.githubusercontent.com/Homebrew/homebrew-dupes/1663b8f234a0b8f6d376572dfe25e6c1f18066d9/rsync.rb
	==> Installing jedda/counterpart/counterpart dependency: https://raw.githubusercontent.com/Homebrew/homebrew-dupes/1663b8f234a0b8f6d376572dfe25e6c1f18066d9/rsync.rb
	==> Downloading https://rsync.samba.org/ftp/rsync/rsync-3.1.1.tar.gz
	Already downloaded: /Library/Caches/Homebrew/rsync-3.1.1.tar.gz
	==> Verifying rsync-3.1.1.tar.gz checksum
	tar xf /Library/Caches/Homebrew/rsync-3.1.1.tar.gz
	==> Downloading https://trac.macports.org/export/127713/trunk/dports/net/rsync/files/patch-fileflags.diff
	/usr/bin/curl -fLA Homebrew 0.9.5 (Ruby 2.0.0-481; OS X 10.9.5) https://trac.macports.org/export/127713/trunk/dports/net/rsync/files/patch-fileflags.diff -C 0 -o /Library/Caches/Homebrew/rsync--patch-b50f0ad6d2c20e561e17b64f07545b1ecfe7d61481a6e5af192abfe21af01e73.diff.incomplete
	  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
	                                 Dload  Upload   Total   Spent    Left  Speed
	  0     0    0     0    0     0      0      0 --:--:--  0:03:28 --:--:--     0curl: (35) Unknown SSL protocol error in connection to trac.macports.org:443 
	Error: Failed to download resource "rsync--patch"
	Download failed: https://trac.macports.org/export/127713/trunk/dports/net/rsync/files/patch-fileflags.diff
	
As far as I see `trac.macports.org` is down?

**Comment by timfenney** [comment]
Yep. That's the new problem.
